File: src/Appartements.java
import java.util.Arrays;
import java.util.Scanner;

public class Appartements {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt(), m = sc.nextInt(), k = sc.nextInt();
        int[] applicants = new int[n], apartements = new int[m];
        for (int i = 0; i < n; i++)
            applicants[i] = sc.nextInt();
        for (int i = 0; i < m; i++)
            apartements[i] = sc.nextInt();
        Arrays.sort(applicants);
        Arrays.sort(apartements);
        int i = 0, j = 0, ans = 0;
        while (i < n && j < m) {
            if (Math.abs(applicants[i] - apartements[j]) <= k) {
                ans++;
                j++;
                i++;
            } else if ((applicants[i] - k) > apartements[j])
                j++;
            else
                i++;
        }
        System.out.println(ans);
    }
}

File: src/appartements.rs
fn main() {
    use std::io::stdin;
    let mut line = String::new();
    stdin().read_line(&mut line);
    let nmk: Vec<i32> = line
        .split_whitespace()
        .filter_map(|w| w.parse().ok())
        .collect();
    let (n, m, k) = (nmk[0], nmk[1], nmk[2]);
    line = "".to_string();
    stdin().read_line(&mut line);
    let mut applicants: Vec<i32> = line
        .split_whitespace()
        .filter_map(|w| w.parse().ok())
        .collect();
    line = "".to_string();
    stdin().read_line(&mut line);
    let mut apartements: Vec<i32> = line
        .split_whitespace()
        .filter_map(|w| w.parse().ok())
        .collect();
    applicants.sort();
    apartements.sort();
    let (mut i, mut j, mut ans) = (0, 0, 0);
    while i < n && j < m {
        if (applicants[i as usize] - apartements[j as usize]).abs() <= k {
            ans += 1;
            j += 1;
            i += 1;
        } else if (applicants[i as usize] - k) > apartements[j as usize] {
            j += 1;
        } else {
            i += 1;
        }
    }
    // println!("{:?}", applicants);
    // println!("{:?}", apartements);
    println!("{}", ans);
}
